src/main.zig

```diff
@@ -3322,6 +3322,46 @@ const AfterUpdateHook = union(enum) {
     update: []const u8,
 };
 
+// we take the raw compile log text, split it back up, and prettify it
+fn printCompileLog(comp: Compilation, ttyconf: std.debug.TTY.Config) !void {
+    const log_text = comp.getCompileLogText();
+    if (log_text.len == 0) return;
+
+    var bw = io.bufferedWriter(io.getStdErr().writer());
+    const stderr = bw.writer();
+
+    ttyconf.setColor(stderr, .Cyan);
+    try stderr.writeAll("\ncompile log output:\n");
+    ttyconf.setColor(stderr, .Reset);
+
+    var log_lines = mem.split(u8, log_text, "\n");
+    while (log_lines.next()) |log_line| {
+        if (log_lines.index == null) break;
+
+        ttyconf.setColor(stderr, .Dim);
+        try stderr.writeAll("- ");
+        ttyconf.setColor(stderr, .Reset);
+
+        var values = mem.split(u8, log_line, "\x00");
+        while (values.next()) |value| {
+            // we bolden the values and dim the commas to
+            // make the values easier to distinguish
+
+            ttyconf.setColor(stderr, .Bold);
+            try stderr.writeAll(value);
+            ttyconf.setColor(stderr, .Reset);
+
+            if (values.index != null) {
```

Comment:
I was wondering if we could add `isLast` to the 3 `split`, `splitBackwards`, and `tokenize` iterators in `std.mem`? Often people (or at least me) try to do something like `iter.index != len - 1` which doesn't actually work because it's an index for the slice, not the iteration index.
I think it's just very common to want to know if this is the last iteration when splitting. Is this something worth adding? I don't think it's obvious what `index` exactly is and that it's `null` on the last iteration. That's why we could add a method  for it.